addons/survey/controllers/main.py

```diff
@@ -433,11 +432,9 @@ def survey_get_question_image(self, survey_token, answer_token, question_id, sug
                           .suggested_answer_ids.filtered(lambda a: a.id == suggested_answer_id):
             return werkzeug.exceptions.NotFound()
 
-        status, headers, image_base64 = request.env['ir.http'].sudo().binary_content(
-            model='survey.question.answer', id=suggested_answer_id, field='value_image',
-            default_mimetype='image/png')
-
-        return request.env['ir.http']._content_image_get_response(status, headers, image_base64)
+        record = request.env['survey.question.answer'].sudo().browse(suggested_answer_id)
```

Comment:
Seems redundant: rather than fiscarding the answer to the conditional at line 431, why not store it in a local then create the stream from that?

Though the deref through `question_ids` and `suggestion_answer_ids` also sewms a bit complicated? @dbeguin @awa-odoo was there a specific reason not to use `search` to find the answer, rather than go through 2 O2Ms?

Reply:
I guess we could have done something like:

```py
suggested_answer = False
if question_id in survey_sudo.question_ids.ids:
    suggested_answer = request.env['survey.question.answer'].sudo().search([
        ('id', '=', suggested_answer_id),
        ('question_id', '=', question_id),
    ])

if not suggested_answer:
    return werkzeug.exceptions.NotFound()
```

Would probably be more performant.
Is that what you meant?

Reply:
Something along those lines yeah, possibly even with a

    ('question_id.survey_id', '=', survey_sudo.id)

or something along those lines rather than the check through `question_ids`.

Less of a performance thing, and more of an avoiding the series of filters which feel a bit meh, I wouldn't expect this to be much of an issue unless somebody creates a survey with a huge number of questions where the matching question would have a huge number of answers.